command.go

```diff
@@ -404,3 +408,29 @@ func hasCommand(commands []*Command, command *Command) bool {
 
 	return false
 }
+
+func checkDuplicatedCmds(parent *Command) error {
+	if parent == nil {
+		return nil
+	}
+	if len(parent.Subcommands) == 0 {
+		return nil
+	}
+
+	var (
+		seen = make(map[string]struct{})
+	)
+	for _, c := range parent.Subcommands {
+		if c == nil {
+			continue
+		}
```

Comment:
What is the purpose of `nil` subcommands here?

Reply:
I'm not sure if we will have nil subcommand situation. I make it more defensive.

Reply:
If `nil` is an allowed subcommand, it should be documented here. Otherwise the code is dead.

https://github.com/urfave/cli/blob/b686f4fd3343d353d0f079f6ce620850e23d329d/command.go#L39-L40

Reply:
```golang
func Test_checkDuplicatedCmds_nilInput(t *testing.T) {
	expect(t, checkDuplicatedCmds(nil), nil)
	cmd := Command{
		Name:        "",
		Subcommands: []*Command{nil},
	}
	expect(t, checkDuplicatedCmds(&cmd), nil)
}
```
In this test, it will panic if we don't check nil pointer.